packages/salesforcedx-vscode-apex/src/commands/forceLaunchApexReplayDebuggerWithCurrentFile.ts

```diff
@@ -81,13 +93,11 @@ async function getApexTestClassName(
   }
 
   await testOutlineProvider.refresh();
-  let testClassName = testOutlineProvider.getTestClassName(sourceUri);
-  // This is a little bizarre.  Intellisense is reporting that getTestClassName() returns a string,
-  // but it actually it returns string | undefined.  Well, regardless, since flushFilePath() takes
-  // a string (and guards against empty strings) using the Non-null assertion operator
-  // (https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator)
-  // fixes the issue.
-  testClassName = fileUtils.flushFilePath(testClassName || '');
+
+  const filePath = sourceUri.toString();
+  // filePath is in the format of "file:///Users/{user-name}/{path-to-apex-file.cls}"
+
+  const testClassName = testOutlineProvider.getTestClassName(filePath);
```

Comment:
So the root of the issue here was flushFilePath throwing an error correct?  I was trying to wrap my head around this change and didn't really grok the root problem.  

One big downside to the change is we are switching from a single lookup in a map to iteration over all tests that exist to see if any match (with an extra operation to lowercase the path for each test).  Granted for small use cases this probably doesn't matter, but we support larger orgs that have hunderds of not thousands of tests so the cost here is potentially high. 

An alternate approach would be to move the flushFilePath call to before the getTestClassName call 
```
  const flushedPath =  fileUtils.flushFilePath(sourceUri);
  const testClassName = testOutlineProvider.getTestClassName(flushedPath);
```

To verify this I wanted to see what was happening with flushFilePath.  It appears there have been issues with the version of node used by VSCode (which explains why we didn't see this originally but it's showing up now) and the usage of `realpathSync.native` [ref](https://github.com/microsoft/vscode-eslint/issues/1229#issuecomment-814057935).

Reply:
We may want to add a try/catch to `flushFilePath`.  I did a quick search and it appears that a pretty typical approach: [example](https://github.dev/facebook/jest/blob/80cf80055682919251b9567ef482755d5f060d47/packages/jest-util/src/tryRealpath.ts#L12%23L12)

Thoughts?

Reply:
Agree with what Gordon has said so far here. Re: iterating over all test files within getTestClassName(), what do you think about lowercasing the `testIndex `entries when it is built in `createTestIndex()` and then lowercase the uri filename when checking the index ` this.testIndex.get(uri.toString().toLowerCase());`?

Reply:
yeah I'm not a fan of unhandled exceptions myself...but I do  trust some of the larger projects have flushed out the details around the native method and I'd rather lean on their experience then learn all the quirks of that method ourselves.  I wish I could just share the link to the real work examples I'm seeing in vscode but alas it doesn't work that way. I see this by hovering over the native method and clicking the 'See Real World example from Github' link. 
![Screenshot 2023-02-21 at 5 00 06 PM](https://user-images.githubusercontent.com/76090802/220478155-9044eefa-2358-4811-a6ff-618db02b4e23.png)

[Typescript](https://github.dev/microsoft/TypeScript/blob/70585aa0c0e4fbd2921f340f276d87fef8801997/src/compiler/sys.ts#L1852%23L1852)
[Jest](https://github.dev/facebook/jest/blob/80cf80055682919251b9567ef482755d5f060d47/packages/jest-util/src/tryRealpath.ts#L12%23L12)
[eslint limits it to specific os](https://github.dev/microsoft/vscode-eslint/blob/1e44f688319912a3671e7070187f2fc7a5430ea1/server/src/eslintServer.ts#L825%23L825)
[opensumi (some framework to build vscode extenions 🤷 )](https://github.dev/opensumi/core/blob/574ebf8423f0fa8c6a7da2be90e3fa1a246e9e2c/packages/file-service/src/node/file-service-watcher.ts#L321%23L321)

There's a few others and not all swallow the exception but those that I looked at appears to be filtering by os and other method prior to calling the native method.  

I'd totally be down to have a catch that posts a custom event to our analytics with the path so we could try and track down when the native method throws. (we could also add a console.log so we could easily identify it's happening with the end user).